src/components/PeopleTable.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useState } from 'react';
+import cn from 'classnames';
+import { Link, useParams } from 'react-router-dom';
+import { Loader } from './Loader';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { PersonLink } from './PersonLink';
+
+interface Parent {
+  name: string;
+  slug: string;
+}
+
+export const PeopleTable = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const { slug } = useParams();
+  let parents: Parent = {
+    name: '',
+    slug: '',
+  };
+
+  useEffect(() => {
+    const fetchPeople = async () => {
+      setIsLoading(true);
+      try {
+        const fetchedPeople = await getPeople();
+
+        setPeople(fetchedPeople);
+        setIsErrorShown(false);
+      } catch {
+        setIsErrorShown(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchPeople();
+  }, []);
+
+  function doesThePersonExist(name: string) {
+    const parent: Person | undefined = people.find(
+      person => person.name === name,
+    );
+
+    if (parent) {
+      parents = {
+        name: parent.name,
+        slug: parent.slug,
+      };
+
+      return true;
+    }
+
+    return false;
+  }
```

Comment:
looks a little bit strange, you just try to find the mother and father using the find method in the map, without this logic with function and vars outside the function